### PR TITLE
fix(core): Fix pairedItem issue with partial manual executions

### PR DESCRIPTION
**Pull request by janober**
## Summary
Fixes a bug with manual partial manual executions with nodes that come after nodes with multiple outputs.

## Related tickets and issues
https://linear.app/n8n/issue/PAY-1398/review-add-test-for-paireditem-fix
https://linear.app/n8n/issue/PAY-1380/bug-dollar-is-broken-on-partial-execution-but-works-on-full-exec

## Review / Merge checklist
- [x] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

File: cypress/e2e/19-execution.cy.ts [hunk 1]
```diff
@@ -569,4 +569,27 @@ describe('Execution', () => {
 			expect(interception.request.body.runData).to.include.all.keys(expectedRunDataKeys);
 		});
 	});
+
+	it('should successful execute partial executions with nodes attached to the second output', () => {
```
Comment (despairblue) [inline cypress/e2e/19-execution.cy.ts:573]: Not the best name, happy to take suggestions. Anything else I come up with is equally convoluted. 😅

Comment (janober) [reply cypress/e2e/19-execution.cy.ts:573]: I totally feel with you. Programming is easy, but naming things is hard.

Comment (despairblue) [reply cypress/e2e/19-execution.cy.ts]: I fixed the adverb at least 😅

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.30.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.30.1)

**Pull request merged**